CBLClient/Apps/CBLTestServer-C/src/CollectionMethods.cpp

```diff
@@ -0,0 +1,387 @@
+            if(err->code!=0)
+                write_serialized_body(conn, CBLError_Message(err));
+            else
+                write_serialized_body(conn, createValueIndex);
+        });
+    }
+
+    //create full text index
+    void collection_createFullTextIndex(json& body, mg_connection* conn) {
+        with<CBLCollection *>(body, "collection", [conn,&body](CBLCollection* collection) {
+            const auto name = flstr(body["name"]);
+            CBLError *err = new CBLError();
+            CBLFullTextIndexConfiguration *config = new CBLFullTextIndexConfiguration();
+            config->expressionLanguage = kCBLN1QLLanguage;
+            config->expressions = flstr(body["expression"]);
+            config->language = flstr(body["language"]);
+            if(body["ignoreAccents"] == true)
+                config->ignoreAccents = true;
+            bool createFullTextIndex = CBLCollection_CreateFullTextIndex(collection, name, *config, err);
+            if(err->code!=0)
+                write_serialized_body(conn, CBLError_Message(err));
+            else
+                write_serialized_body(conn, createFullTextIndex);
+        });
+    }
+
+    //delete index
+    void collection_deleteIndex(json& body, mg_connection* conn) {
+        with<CBLCollection *>(body, "collection", [conn, &body](CBLCollection* collection) {
+            auto name = body["name"];
+            CBLError* err = new CBLError();
+            auto deleteIndex = CBLCollection_DeleteIndex(collection, flstr(name), err);
+            if(err->code!=0)
+                write_serialized_body(conn, CBLError_Message(err));
+            else
+                write_serialized_body(conn, deleteIndex);
+        });
+    }
+
+    //get index names
+    void collection_getIndexNames(json& body, mg_connection* conn) {
+        with<CBLCollection *>(body, "collection", [conn, &body](CBLCollection* collection) {
+            json keys(0, nullptr);
+            CBLError* err = new CBLError();
+            FLMutableArray index_names = CBLCollection_GetIndexNames(collection, err);
+            if(err->code!=0)
+                write_serialized_body(conn, CBLError_Message(err));
+            else {
+                FLArrayIterator i;
+                FLArrayIterator_Begin(index_names, &i);
+                do{
+                    const auto indexName = FLArrayIterator_GetValue(&i);
+                    FLStringResult jsonString = FLValue_ToString(indexName);
+                    keys.push_back(jsonString);
+                } while(FLArrayIterator_Next(&i));
+                write_serialized_body(conn, keys);
+            }
+        });
+    }
+
+    //add change listener
+    void collection_addChangeListener(json& body, mg_connection* conn) {
+        with<CBLCollection *>(body, "collection", [conn](CBLCollection* collection) {
+            CBLListenerToken* token = CBLCollection_AddChangeListener(collection, CBLCollection_DummyListener, nullptr);
```

Comment:
As DummyListener doesn't seem to do anything, will there be more work on this?

Reply:
I am still exploring the context.